morango/models.py

```diff
@@ -69,18 +69,26 @@ class InstanceIDModel(UUIDModelMixin):
     counter = models.IntegerField(default=0)
     current = models.BooleanField(default=True)
     db_path = models.CharField(max_length=1000)
+    system_id = models.CharField(max_length=100, blank=True)
 
     @staticmethod
     def get_or_create_current_instance():
         """Get the instance model corresponding to the current system, or create a new
         one if the system is new or its properties have changed (e.g. OS from upgrade)."""
 
+        # on Android, platform.platform() barfs, so we handle that safely here
+        try:
+            plat = platform.platform()
+        except:
```

Comment:
The exception is pretty generic, and also doesn't make much sense for the context (something like "IOError: this directory is a folder"), so I don't think it would add much semantic value. And if something as simple as `platform.platform()` fails, no matter what the reason, I think we'd want to just handle it gracefully anyway.